src/Integration.UnitTests/TeamExplorer/SectionControllerTests.cs

```diff
@@ -0,0 +1,290 @@
+        public void SectionController_BrowseToUrlCommand()
+        {
+            // Setup
+            var webBrowser = new ConfigurableWebBrowser();
+            var testSubject = this.CreateTestSubject(webBrowser);
+
+            // Case 1: Empty URL
+            // Act + Verify CanExecute
+            Assert.IsFalse(testSubject.BrowseToUrlCommand.CanExecute(null));
+
+            // Case 2: Bad URL
+            // Act + Verify CanExecute
+            Assert.IsFalse(testSubject.BrowseToUrlCommand.CanExecute("not a Uri"));
+
+            // Case 3: Good URL
+            const string goodUrl = "http://localhost";
+
+            // Act + Verify CanExecute
+            Assert.IsTrue(testSubject.BrowseToUrlCommand.CanExecute(goodUrl));
+
+            // Act + Verify Execute
+            testSubject.BrowseToUrlCommand.Execute(goodUrl);
+            webBrowser.AssertNavigateToCalls(1);
+            webBrowser.AssertRequestToNavigateTo(goodUrl);
+        }
+
+        #endregion
+
+        #region Helpers
+        private static void ReInitialize(SectionController controller)
+        {
+            controller.Host.ClearActiveSection();
+            controller.Host.VisualStateManager.ManagedState.ConnectedServers.Clear();
+            controller.Initialize(null, new Microsoft.TeamFoundation.Controls.SectionInitializeEventArgs(new ServiceContainer(), null));
+            bool refreshCalled = false;
+            controller.RefreshCommand = new RelayCommand(() => refreshCalled = true);
+            controller.Refresh();
+            Assert.IsTrue(refreshCalled, "Refresh command execution was expected");
+        }
+
+        private class TestCommandTarget : IOleCommandTarget
+        {
+            private int queryStatusNumberOfCalls;
+
+            #region IOleCommandTarget
+            int IOleCommandTarget.Exec(ref Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
+            {
+                throw new NotImplementedException();
+            }
+
+            int IOleCommandTarget.QueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[] prgCmds, IntPtr pCmdText)
+            {
+                this.queryStatusNumberOfCalls++;
+                return this.QueryStatusReturnsResult;
+            }
+            #endregion
+
+            #region  Test helpers
+            public int QueryStatusReturnsResult
+            {
+                get;
+                set;
+            } = SectionController.CommandNotHandled;
+
+            public void AssertQueryStatusCalled(int expectedNumberOfTimes)
+            {
+                Assert.AreEqual(expectedNumberOfTimes, this.queryStatusNumberOfCalls, "IOleCommandTarget.QueryStatus is called unexpected number of times");
+            }
+            #endregion
+        }
+
+        private SectionController CreateTestSubject(IWebBrowser webBrowser = null)
+        {
+            var host = new ConfigurableHost(this.serviceProvider, Dispatcher.CurrentDispatcher);
```

Comment:
Related to my other comment about `internal IHost` on `SectionController`, we could have this `ConfigurableHost` created at test init and then referenced here & in `ReInitialize`; removing the need for the `internal` access modifier.